src/sentry/spans/consumers/recombine/factory.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import random
+from collections.abc import Mapping
+
+import sentry_sdk
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies.abstract import ProcessingStrategy, ProcessingStrategyFactory
+from arroyo.processing.strategies.commit import CommitOffsets
+from arroyo.processing.strategies.run_task import RunTask
+from arroyo.types import BrokerValue, Commit, Message, Partition
+
+from sentry.spans.consumers.recombine.message import process_segment
+from sentry.utils import json
+
+logger = logging.getLogger(__name__)
+
+
+def process_message(message: Message[KafkaPayload]):
+    assert isinstance(message.value, BrokerValue)
+    try:
+        segments = json.loads(message.payload.value)
```

Comment:
Will update this to validate schema once it's available, probably as a follow up